code/Calendar.php

```diff
@@ -276,19 +276,31 @@ public static function getWeekdayName($value, $width = 'wide', $locale = '', $st
     {
         $result = '';
         if ((!empty($value)) || ($value === 0) || ($value === '0')) {
-            $weekday = null;
-            if (is_int($value)) {
-                $weekday = $value;
-            } elseif (is_float($value)) {
-                $weekday = intval($value);
-            } elseif (is_string($value)) {
-                if (is_numeric($value)) {
+            switch (gettype($value)) {
```

Comment:
Mhm… I believe the PHP documentation states that the return type of this method is subject to change and thus the code may break in newer PHP versions.

I'd certainly prefer to just create multiple methods for each datatype.